config/config.go

```diff
@@ -908,6 +935,18 @@ func GetPackageReference(path string) *PackageReference {
 	return &PackageReference{Package: match[1], PathInPackage: match[2]}
 }
 
+// GetPackageDirectoryFromType determines the viam subdirectories for packages based on type.
+func GetPackageDirectoryFromType(packageType PackageType) string {
+	switch packageType {
+	case PackageTypeMlModel:
+		return "ml_models"
+	case PackageTypeModule:
+		return "modules"
+	default:
+		return ""
```

Comment:
the reasoning here is that we don't currently have types for packages, so this is to keep backward compatibility. When that migration happens, however, i agree we should implement something like this.

Reply:
Why not just string(PackageType)? This will eliminate the number of places we have to update when a new type is added.

To James's point, I think that for now a default of an empty/nil package type should result in "ml_models" as default. Later in the project we will have a migration to convert all robot configs to have the `"type" : "ml_models"` so that it is explicit.

Reply:
package_type is singular whereas the subdirectories are plural (not sure if thats important but that's how its currently done and what makes grammatical sense for file structure). 

I agree with adding a something else as a default, but if we do that now it may break packages as there is no type and not sure if symlink for packages will actually work if we do that. (something that I am planning to test). I can add a ticket to do this though once we do the migration.